### PR TITLE
Standardized formatting with `black`

### PR DESCRIPTION
**Pull request by BSchilperoort**
The code has been formatted using [black](https://github.com/psf/black). Formatting code using `black` can be useful because:

> Blackened code looks the same regardless of the project you're reading. Formatting becomes transparent after a while and you can focus on the content instead.
> 
> Black makes code review faster by producing the smallest diffs possible.

I have found this to be very useful in, for example, [s2spy](https://github.com/AI4S2S/s2spy).

Additionally:

- the cdsapi version has been updated to the latest stable version
- The codecov badge now correctly refers to `main` (instead of `master`).
- The version string in `conf.py` was previously not updated, it is now updated to the current version
- String formatting has been updated to use so-called f-strings (see [PEP498](https://peps.python.org/pep-0498/))
- Added an extra test to test_info, to ensure that the right error is raised when calling, e.g., `era5cli info "blabla"`
- Added black & isort to both the CI and the contributing instructions
- flake8 is now configured through the `flake8-pyproject` plugin

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -12,36 +12,39 @@
...
-project = 'era5cli'
-copyright = '2019, Netherlands eScience Center'
-author = 'era5cli team'
+project = "era5cli"
+copyright = "2019, Netherlands eScience Center"
+author = "era5cli team"
 
 # The full version, including alpha/beta/rc tags
-release = '1.3.1'
+release = "1.3.2"
```
Comment (bvreede) [inline docs/source/conf.py:29]: I realize I missed this with the recent new release! Perhaps it's good to update the contribution guide as well; adding conf.py to the documents that need to be updated when  bumping the version (and perhaps remove outdated information from that list, as some docs may not exist anymore as of #134).

File: era5cli/utils.py [hunk 2]
```diff
@@ -108,15 +105,13 @@ def _format_hours(values: list) -> list:
...
-            'invalid value specified for hours: {}'.format(value))
-        returnlist += ["{}:00".format(str(value).zfill(2))]
+        assert int(value) >= 0, f"invalid value specified for hours: {value}"
+        assert int(value) <= 23, f"invalid value specified for hours: {value}"
+        returnlist += [f"{str(value).zfill(2)}:00"]
     return returnlist
 
 
-def _print_multicolumn(header: str, info: list):
+def print_multicolumn(header: str, info: list):
```
Comment (bvreede) [inline era5cli/utils.py:114]: Is this the formatter's choice or yours? It's not clear to me why some function names were changed from `_function_name()` to `function_name()` and others not.

**Review by bvreede — APPROVED** [review]
Noice!

File: era5cli/utils.py (hunk 2 above)
Comment (BSchilperoort) [reply era5cli/utils.py:114]: I changed function definitions where they had the underscore, but were used in different modules. The underscore generally denotes "internal use". `_zpadlist` in `utils`, for example, is only used within utils.py. However, `print_multicolumn` is used outside of utils, and should therefore not have an underscore.

File: docs/source/conf.py (hunk 1 above)
Comment (BSchilperoort) [reply docs/source/conf.py:29]: Good point, I'll add it to the contributing document.

**Pull request merged**